app/src/main/java/com/duckduckgo/app/systemsearch/SystemSearchViewModel.kt

```diff
@@ -328,25 +330,38 @@ class SystemSearchViewModel @Inject constructor(
         }
     }
 
-    private fun deleteQuickAccessItem(savedSite: SavedSite) {
+    fun deleteQuickAccessItem(savedSite: SavedSite) {
         when (savedSite) {
             is SavedSite.Favorite -> {
                 viewModelScope.launch(dispatchers.io() + NonCancellable) {
                     savedSitesRepository.delete(savedSite)
                 }
             }
+
             else -> throw IllegalArgumentException("Illegal SavedSite to delete received")
         }
     }
 
-    fun insertQuickAccessItem(savedSite: SavedSite) {
-        when (savedSite) {
-            is SavedSite.Favorite -> {
-                viewModelScope.launch(dispatchers.io()) {
-                    savedSitesRepository.insert(savedSite)
-                }
-            }
-            else -> throw IllegalArgumentException("Illegal SavedSite to delete received")
+    private fun hideQuickAccessItem(quickAccessFavourite: FavoritesQuickAccessAdapter.QuickAccessFavorite): Int {
+        val quickAccessItems = resultsViewState.value as QuickAccessItems
+        val favourites = quickAccessItems.favorites.toMutableList()
+        val index = favourites.indexOf(quickAccessFavourite)
+        favourites.remove(quickAccessFavourite)
+
+        latestQuickAccessItems = QuickAccessItems(favourites)
+        resultsViewState.postValue(latestQuickAccessItems)
```

Comment:
same as commented in `BrowserTabViewmodel` should work. I would keep the observable approach, and avoid updating the state from here.